### PR TITLE
Make sure to only update valid vehicles

### PR DESCRIPTION
**Pull request by janisozaur**
Lets us load parks like Six Flags Carolina:
http://www.nedesigns.com/park/1571/six-flags-carolina/

See https://www.reddit.com/r/openrct2/comments/4g748k/not_able_to_open_certain_parks/

### CODE REVIEW

File: src/world/map.c [hunk 1]
```diff
@@ -2235,7 +2243,7 @@ static int map_get_corner_height(rct_map_element *mapElement, int direction)
 static money32 smooth_land_tile(int direction, uint8 flags, int x, int y, int targetBaseZ, int minBaseZ)
 {
 	// Check if inside map bounds
-	if (x < 0 || y < 0 || x >= (256 * 32) || y >= (256 * 32)) {
+	if (!map_is_location_valid(x, y)) {
```
Comment (IntelOrca) [inline src/world/map.c]: The logic in your function is different to this as `MAP_MINIMUM_X_Y` is -256.

Comment (janisozaur) [inline src/world/map.c]: Yes, I'm aware and I was just about to point this out.

What is `MAP_MINIMUM_X_Y`? I can do `<=` so the functionality will stay the same, but which one is correct?

**Comment by janisozaur** [comment]
Also note: this park has abysmal performance in OpenRCT2, not sure about vanilla.

File: src/world/map.c (hunk 1 above)
Comment (IntelOrca) [inline src/world/map.c]: No idea

Comment (janisozaur) [inline src/world/map.c]: I'll change it to be `<=`, just for the sake of keeping it the way it is now…

**Comment by janisozaur** [comment]
Updated.

File: src/world/map.c [hunk 2]
```diff
@@ -646,6 +646,14 @@ int map_height_from_slope(int x, int y, int slope)
 	return 0;
 }
 
+int map_is_location_valid(int x, int y)
```
Comment (IntelOrca) [inline src/world/map.c]: Use `bool` instead?

Comment (janisozaur) [inline src/world/map.c]: done

**Pull request merged**